modules/apps/layout/layout-js-components-web/src/main/resources/META-INF/resources/js/components/modals/MarketplaceViews.tsx

```diff
@@ -0,0 +1,181 @@
+/* eslint-disable no-console */
+
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import {
+	Cart,
+	Marketplace,
+	MarketplaceRest,
+	MarketplaceView,
+	Product,
+	useMarketplaceContext,
+} from '@liferay/marketplace-js-components-web';
+import {fetch} from 'frontend-js-web';
+import React from 'react';
+
+import {InstallFragmentModalBody} from './InstallFragmentModal';
+
+async function fetchFragmentBlob(marketplaceRest: MarketplaceRest, url: URL) {
+	const response = await marketplaceRest.fetchMarketplace<Response>(
+		url.pathname,
+		{
+			earlyReturn: true,
+		}
+	);
+
+	return response.blob();
+}
+
+async function getVirtualEntryBlob(
```

Comment:
I see getVirtualEntryBlob isn't being used, so SF would complain.

Can I comment it for the moment?

Reply:
Yea!

Ideally we will only use this approach to get the blob from the virtual entry (order) but due the product limitation we will workaround using the attachment while we find the final solution

Reply:
Worst case scenario you can delete and we can bring it back later

Reply:
Good to know, Just so I know, would you merge this or is this only a poc?

Reply:
I'm not going to merge this one, is just part of the POC to help your guys